Channel Sidebar Organization

devinbinnie opened the pull request:
#### Summary
Merging Phase 1 of Channel Sidebar Organization into master.
Features include:
- Collapsable categories
- Unread filter
- History arrows for the desktop
- Moved channel switcher

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-21296

#### Related PRs
Server: https://github.com/mattermost/mattermost-server/pull/14036
Redux: https://github.com/mattermost/mattermost-redux/pull/1089

mattermod commented:
Creating a new SpinWick test server using Mattermost Cloud.

mattermod commented:
Mattermost test server created! :tada:

Access here: https://mattermost-webapp-pr-4999.test.mattermost.cloud

| Account Type | Username | Password |
|---|---|---|
| Admin | sysadmin | Sys@dmin123 |
| User | user-1 | User-1@123 |

mattermod commented:
New commit detected. SpinWick will upgrade if the updated docker image is available.

mattermod commented:
Mattermost test server updated with git commit `d25123d624e0bf910d5e4a9b3083ae892744b50d`.

Access here: https://mattermost-webapp-pr-4999.test.mattermost.cloud

devinbinnie commented:
/update-branch

mattermod commented:
Error trying to update the PR.
Please do it manually.

mattermod commented:
New commit detected. SpinWick will upgrade if the updated docker image is available.

mattermod commented:
Test server creation failed. See the logs for more information.

mattermod commented:
New commit detected. SpinWick will upgrade if the updated docker image is available.

mattermod commented:
Mattermost test server updated with git commit `3938a23ecde0d6da5438674d465f0cff0854fa88`.

Access here: https://mattermost-webapp-pr-4999.test.mattermost.cloud

mattermod commented:
New commit detected. SpinWick will upgrade if the updated docker image is available.

mattermod commented:
Mattermost test server updated with git commit `0b7c625a9f011b4aa3f7360cbdbb4b2a0ebdd6db`.

Access here: https://mattermost-webapp-pr-4999.test.mattermost.cloud

mattermod commented:
New commit detected. SpinWick will upgrade if the updated docker image is available.

mattermod commented:
Mattermost test server updated with git commit `c4305b85072c73090e2d0b7585c2a7e8c0eb68ac`.

Access here: https://mattermost-webapp-pr-4999.test.mattermost.cloud

mattermod commented:
New commit detected. SpinWick will upgrade if the updated docker image is available.

mattermod commented:
Mattermost test server updated with git commit `f0d3b7ec46963a3a8e316fc12c29162f28cc31e5`.

Access here: https://mattermost-webapp-pr-4999.test.mattermost.cloud

esethna commented:
1. Private and Faves sections shouldn't exist if there are no channels in it:

![image](https://user-images.githubusercontent.com/14333569/76267203-36bfaf80-6227-11ea-80a9-c3811eab5147.png)

2. DM section should have a plus button in the unread filter state
![image](https://user-images.githubusercontent.com/14333569/76267233-4c34d980-6227-11ea-8a38-078fe64b0d61.png)

3. Opening the "+" button feels chunky
![plus](https://user-images.githubusercontent.com/14333569/76267273-6bcc0200-6227-11ea-9a28-1d25c61adb92.gif)

4. ^ Animation to collapse also feels off (

5. DMs shouldn't have an arrow if there's no DMs
![image](https://user-images.githubusercontent.com/14333569/76267381-c7968b00-6227-11ea-8780-80b6b5727866.png)

6. @Andrew brown how should we deal with sections that have no unread channels when the unreads filter is applied? This feels funky:
![image](https://user-images.githubusercontent.com/14333569/76267424-f14fb200-6227-11ea-9ed9-0aaf731f5a06.png)

7. Archived state
![image](https://user-images.githubusercontent.com/14333569/76267642-aedaa500-6228-11ea-93ea-baac7e79ed13.png)

esethna commented:
As a side note, I'm kind of worried about the usability of the unreads filter right now since you can't expand sections once you're in the unread view. @andrewbrown00 any high level thoughts on how it feels before we dive into the nitty gritty of the UI (after the rest of the functionality is ready)

mattermod commented:
New commit detected. SpinWick will upgrade if the updated docker image is available.

devinbinnie commented:
@esethna 

1. Fixed!
2. Fixed!
3/4. You'll need to be a bit more specific, since we're using the same menu as the channel header and main hamburger menu. One thing I did notice was that the fact that the `z-index` of the category headers is >1 causes them to briefly be on top of the menu for a split second. I'm not sure there's anything we can do about that.
5. Fixed!
6. -
7. Using the new `icon-archive` and `icon-pencil` icons for archive and draft respectively. @andrewbrown00 let me know if I need to change this
8. Fixed!
9. Fixed!

mattermod commented:
Mattermost test server updated with git commit `f1107bffe8cb874b984594cbd0c03d6461d05887`.

Access here: https://mattermost-webapp-pr-4999.test.mattermost.cloud

mattermod commented:
New commit detected. SpinWick will upgrade if the updated docker image is available.

mattermod commented:
New commit detected. SpinWick will upgrade if the updated docker image is available.

mattermod commented:
Test server creation failed. See the logs for more information.

mattermod commented:
Mattermost test server updated with git commit `b2c90db4d8fedd85b87456c8721a74ee97d5489e`.

Access here: https://mattermost-webapp-pr-4999.test.mattermost.cloud

mattermod commented:
New commit detected. SpinWick will upgrade if the updated docker image is available.

mattermod commented:
Mattermost test server updated with git commit `6da9e0557cd6d04a09b0ca66336fefba7276e90c`.

Access here: https://mattermost-webapp-pr-4999.test.mattermost.cloud

mattermod commented:
New commit detected. SpinWick will upgrade if the updated docker image is available.

mattermod commented:
Test server creation failed. See the logs for more information.

mattermod commented:
Test server destroyed

mattermod commented:
Creating a new SpinWick test server using Mattermost Cloud.

mattermod commented:
Test server creation failed. See the logs for more information.

mattermod commented:
Test server destroyed

mattermod commented:
Creating a new SpinWick test server using Mattermost Cloud.

mattermod commented:
Mattermost test server created! :tada:

Access here: https://mattermost-webapp-pr-4999.test.mattermost.cloud

| Account Type | Username | Password |
|---|---|---|
| Admin | sysadmin | Sys@dmin123 |
| User | user-1 | User-1@123 |

mattermod commented:
New commit detected. SpinWick will upgrade if the updated docker image is available.

mattermod commented:
Mattermost test server updated with git commit `58a455e39f28388eee57ba5f6210a4aaafb10be9`.

Access here: https://mattermost-webapp-pr-4999.test.mattermost.cloud

deanwhillier commented:
Working through a review now. Notes will come in in a few posts. I can help work on any of these, just let me know @devinbinnie!

Compare spacing with designs, here's a pic showing some of the differences:

![image](https://user-images.githubusercontent.com/3245614/76380389-0829fd80-6329-11ea-8c83-95edfc7eed04.png)

The thicker horizontal line indicates where I centred both images vertically.

Figma has functionality to help determine distances of objects from each other – if you hold down OPTION with an object selected and then mouse over another object you want to measure the distance to, Figma will overlay the measurement for you.

The UX team is using an 8px based measurement system, so distances are usually multiples of 8.

Note that bounding boxes in design tools and the web unfortunately aren't always the same, so it's not always enough to just set distances between bounding boxes (margins, padding etc.) on the web to get the correct measurements.

Apologies if you are already aware of any/all of the above! 😛

deanwhillier commented:
Double check fonts against the design:

- Categories use `font-size: 12px` with `font-weight: 600` (semi bold).
- Selected/unread channels use `font-weight: 600` (semi bold).
- Chevron used for expanding/collapsing the categories is `font-size: 12px`
- Mention badge number is `font-size: 10px` with `font-weight: 700` (bold). It is challenging to get a number to fit properly within a small circle, so this may need to be tweaked to work on web (this may already have been tried)

deanwhillier commented:
DM labels jump up/down between unread and read states. Looks like the `x` close button may be the cause.

![shifting-channel-text](https://user-images.githubusercontent.com/3245614/76383620-c0a86f00-6332-11ea-9cfb-ce539c082e34.gif)

deanwhillier commented:
When a channel is selected, the icon colour/opacity doesn't change.

![image](https://user-images.githubusercontent.com/3245614/76383688-ff3e2980-6332-11ea-944f-0f915da5dee5.png)

vs design

![image](https://user-images.githubusercontent.com/3245614/76383715-1250f980-6333-11ea-8078-c972f3ad83be.png)

deanwhillier commented:
Public & Private categories should be `Public Channels` and `Private Channels` respectively.

deanwhillier commented:
GM icon with user count should be square:

![image](https://user-images.githubusercontent.com/3245614/76384168-90fa6680-6334-11ea-89e1-3f073eb96c64.png)

vs. design

![image](https://user-images.githubusercontent.com/3245614/76384200-a2437300-6334-11ea-9398-5d8414ffe3c9.png)

deanwhillier commented:
Actually @andrewbrown00, can you please verify my statement above that goes like so when you have a minute?

> When a channel is selected, the icon colour/opacity doesn't change and the text has font-weight: 600.

mattermod commented:
New commit detected. SpinWick will upgrade if the updated docker image is available.

mattermod commented:
Mattermost test server updated with git commit `18d8bbea5817a2b8a1cbdefeed4f558f6ad5eef9`.

Access here: https://mattermost-webapp-pr-4999.test.mattermost.cloud

mattermod commented:
New commit detected. SpinWick will upgrade if the updated docker image is available.

andrewbrown00 commented:
> Actually @andrewbrown00, can you please verify my statement above that goes like so when you have a minute?
> 
> > When a channel is selected, the icon colour/opacity doesn't change and the text has font-weight: 600.

@deanwhillier thanks for the thorough review! To answer your statement, the selected state shouldn't **bold** the channel name unless the channel is `unread`. The current implementation is correct for the selected channel style.

mattermod commented:
Mattermost test server updated with git commit `fee3d3dcee1dc4b66801d5d8b566decbed77cb6d`.

Access here: https://mattermost-webapp-pr-4999.test.mattermost.cloud

mattermod commented:
New commit detected. SpinWick will upgrade if the updated docker image is available.

deanwhillier commented:
> @deanwhillier thanks for the thorough review! To answer your statement, the selected state shouldn't bold the channel name unless the channel is unread. The current implementation is correct for the selected channel style.

Perfect, that's what I was thinking when I looked at it again this morning. 🙂

hmhealey commented:
I found a bug while working on the E2E tests. The icon for a DM with an archived user is incorrect
![Screen Shot 2020-03-11 at 10 37 52 AM](https://user-images.githubusercontent.com/3277310/76428683-60421d80-6384-11ea-95ab-1b66515057de.png)

mattermod commented:
Mattermost test server updated with git commit `ce6c14b01820a2e03feb7a192ea777b02aae7692`.

Access here: https://mattermost-webapp-pr-4999.test.mattermost.cloud

hmhealey commented:
A couple other things I've run into:
- If you leave a channel, it remains in the new sidebar
- If you delete a channel, it remains in the new sidebar

mattermod commented:
New commit detected. SpinWick will upgrade if the updated docker image is available.

devinbinnie commented:
@esethna Fixed 2-4
1) Requires a bunch of design changes, because the category component isn't aware of the workings of the channel component (which controls collapsing, etc.). So I think we should push this until after the PR is merged, and log it as a known issue.

@hmhealey Fixed the icon on DMs, as for the channels not disappearing, should Redux be handling that or do I have to do something specific to alert Redux to check for channel membership/deleted channels?

@deanwhillier Will get to your stuff next :)

mattermod commented:
Mattermost test server updated with git commit `002a8a68551e8369150d6aa25dfd8101340d2861`.

Access here: https://mattermost-webapp-pr-4999.test.mattermost.cloud

mattermod commented:
New commit detected. SpinWick will upgrade if the updated docker image is available.

mattermod commented:
New commit detected. SpinWick will upgrade if the updated docker image is available.

mattermod commented:
Test server creation failed. See the logs for more information.

mattermod commented:
Mattermost test server updated with git commit `e1b4a24f39a13a807593eaeb0c964b6ce62239b6`.

Access here: https://mattermost-webapp-pr-4999.test.mattermost.cloud

esethna commented:
![image](https://user-images.githubusercontent.com/14333569/76454886-7f47ac00-6392-11ea-850a-532d7a5df526.png)

1. Category headers should be aligned with unread button and jump to
2. Faves category is incorrect font size
3. DMs + button is not aligned

esethna commented:
@andrewbrown00 do you have the expected for the hover state on the jump to?
![image](https://user-images.githubusercontent.com/14333569/76455361-e6fdf700-6392-11ea-801c-462c29123e3c.png)

devinbinnie commented:
> ![image](https://user-images.githubusercontent.com/14333569/76454886-7f47ac00-6392-11ea-850a-532d7a5df526.png)
> 
> 1. Category headers should be aligned with unread button and jump to
> 2. Faves category is incorrect font size
> 3. DMs + button is not aligned

2. As per the spec, the 'Favorites' category is supposed to have a larger font. @andrewbrown00 is this an oversight or intentional?

esethna commented:
Oversight ^

mattermod commented:
New commit detected. SpinWick will upgrade if the updated docker image is available.

mattermod commented:
Mattermost test server updated with git commit `e4307ad3753e31c95083a3e530ae761c942a812c`.

Access here: https://mattermost-webapp-pr-4999.test.mattermost.cloud

mattermod commented:
New commit detected. SpinWick will upgrade if the updated docker image is available.

mattermod commented:
Mattermost test server updated with git commit `1998d7d5e05db85940c3392c417fafbcf214361f`.

Access here: https://mattermost-webapp-pr-4999.test.mattermost.cloud

mattermod commented:
New commit detected. SpinWick will upgrade if the updated docker image is available.

mattermod commented:
Mattermost test server updated with git commit `3a489c2b3fb4dffd22f171ce6bc3243579bd5413`.

Access here: https://mattermost-webapp-pr-4999.test.mattermost.cloud

andrewbrown00 commented:
> @andrewbrown00 do you have the expected for the hover state on the jump to?
> ![image](https://user-images.githubusercontent.com/14333569/76455361-e6fdf700-6392-11ea-801c-462c29123e3c.png)

The style for the hover state has on overlay of `background: rgba(255, 255, 255, 0.08);` and bumps up the text to `color: #FFFFFF;` on top of the default control which renders as: 
![image](https://user-images.githubusercontent.com/48559658/76530826-abbeff00-644a-11ea-96bc-a5147283fca6.png)

andrewbrown00 commented:
> > ![image](https://user-images.githubusercontent.com/14333569/76454886-7f47ac00-6392-11ea-850a-532d7a5df526.png)
> > 
> > 1. Category headers should be aligned with unread button and jump to
> > 2. Faves category is incorrect font size
> > 3. DMs + button is not aligned
> 
> 1. As per the spec, the 'Favorites' category is supposed to have a larger font. @andrewbrown00 is this an oversight or intentional?

This was a synching issue in Figma, not sure why it didn't auto updated like the other category headers that share the style. It has been resolved.
![image](https://user-images.githubusercontent.com/48559658/76531311-66e79800-644b-11ea-9f02-077bee864307.png)

mattermod commented:
New commit detected. SpinWick will upgrade if the updated docker image is available.

mattermod commented:
Mattermost test server updated with git commit `68a989bc2a51657b9fe93859b17e1dcee0a12c65`.

Access here: https://mattermost-webapp-pr-4999.test.mattermost.cloud

mattermod commented:
New commit detected. SpinWick will upgrade if the updated docker image is available.

mattermod commented:
Mattermost test server updated with git commit `0d431991abbf2013594c6beb5eb125fd9fba63ce`.

Access here: https://mattermost-webapp-pr-4999.test.mattermost.cloud

devinbinnie commented:
@esethna All should be fixed except for the arrow issue, which we'll need an updated font from @andrewbrown00. He's working on it now.

mattermod commented:
New commit detected. SpinWick will upgrade if the updated docker image is available.

mattermod commented:
Mattermost test server updated with git commit `e398f24a67ed8572897a8f95041c2e45fa49e9a5`.

Access here: https://mattermost-webapp-pr-4999.test.mattermost.cloud

mattermod commented:
New commit detected. SpinWick will upgrade if the updated docker image is available.

mattermod commented:
Mattermost test server updated with git commit `fbbd263d38da814b3f115f6aa8142e78c467757e`.

Access here: https://mattermost-webapp-pr-4999.test.mattermost.cloud

mattermod commented:
New commit detected. SpinWick will upgrade if the updated docker image is available.

mattermod commented:
Mattermost test server updated with git commit `342038b2d316cb94465986642e19cdbc9f0efdb8`.

Access here: https://mattermost-webapp-pr-4999.test.mattermost.cloud

mattermod commented:
New commit detected. SpinWick will upgrade if the updated docker image is available.

jgilliam17 commented:
Thanks @devinbinnie 

1. I am not seeing history arrows on the test server
2. Sidebar learn more and give feedback links are `not found`
3. Webapp struggles to reload/refresh the page when exp. sidebar feature is ON, page stops responding
4. While feature is ON, and user opens account settings menu or status menu, background theme can be seen/flashes on the menu. see gif

![Menu background theme flash](https://user-images.githubusercontent.com/52937121/76635537-01b0a700-651e-11ea-976f-68460dbbb67c.gif)

mattermod commented:
Mattermost test server updated with git commit `96c55aa94eed538373542c082183b0267ccd13c2`.

Access here: https://mattermost-webapp-pr-4999.test.mattermost.cloud

devinbinnie commented:
> Thanks @devinbinnie
> 
> 1. I am not seeing history arrows on the test server

These are only visible on the Desktop App, so you'll have to test from there.

> 2. Sidebar learn more and give feedback links are `not found`

They haven't been set up yet, I think @esethna is handling that.

> 3. Webapp struggles to reload/refresh the page when exp. sidebar feature is ON, page stops responding

I can't reproduce this, can anyone else?

> 4. While feature is ON, and user opens account settings menu or status menu, background theme can be seen/flashes on the menu. see gif
> 
> ![Menu background theme flash](https://user-images.githubusercontent.com/52937121/76635537-01b0a700-651e-11ea-976f-68460dbbb67c.gif)

Yes this is very annoying, it has to do with the fact that the headers for the categories are "sticky" (meaning that as you scroll up, you can still see the header title while viewing channels in that category). Unfortunately the browser seems to take a small fraction of time to process that and so it thinks briefly that the dropdown menu should be below the headers for a second. I don't think there's any way to fix this right now, but we should make a separate ticket to investigate later on.

jgilliam17 commented:
@devinbinnie Here's an error when reloading the page
<img width="1482" alt="Screen Shot 2020-03-13 at 11 52 17 AM" src="https://user-images.githubusercontent.com/52937121/76637703-99fc5b00-6521-11ea-81f6-06c72cea506e.png">

devinbinnie commented:
> @devinbinnie Here's an error when reloading the page
> <img alt="Screen Shot 2020-03-13 at 11 52 17 AM" width="1482" src="https://user-images.githubusercontent.com/52937121/76637703-99fc5b00-6521-11ea-81f6-06c72cea506e.png">

Looks like it's not loading the package correctly for some reason. Can you try to Empty Cache and Hard Reload?

jgilliam17 commented:
I did both, and still see the errors on Chrome
On Firefox, I don't see any errors in the dev tools, but there is a yellow banner `a webpage is slowing down your browser. what would you like to do?` Stopping it spits out a long row of NetworkErrors
<img width="1173" alt="Screen Shot 2020-03-13 at 12 07 48 PM" src="https://user-images.githubusercontent.com/52937121/76638730-5276ce80-6523-11ea-8c9e-880f92434bc6.png">

devinbinnie commented:
> I did both, and still see the errors on Chrome
> On Firefox, I don't see any errors in the dev tools, but there is a yellow banner `a webpage is slowing down your browser. what would you like to do?` Stopping it spits out a long row of NetworkErrors
> <img alt="Screen Shot 2020-03-13 at 12 07 48 PM" width="1173" src="https://user-images.githubusercontent.com/52937121/76638730-5276ce80-6523-11ea-8c9e-880f92434bc6.png">

That's really strange, I don't think that's normal behaviour, have you tried another machine?

deanwhillier commented on components/sidebar/sidebar_channel/sidebar_group_channel/sidebar_group_channel.tsx:
```diff
@@ -0,0 +1,65 @@
+        const currentUserId = this.props.currentUserId;
+        this.props.actions.savePreferences(currentUserId, [{user_id: currentUserId, category, name: id, value: 'false'}]).then(callback);
+
+        trackEvent('ui', 'ui_direct_channel_x_button_clicked');
```
Just double checking, is this supposed to be 'ui_**direct**_channel_x_button_clicked' instead of **group**?

devinbinnie replied:
It matches what's in the original sidebar, so I kept it that way. We can change it up if you'd like.

deanwhillier replied:
Seems like it should be **group** given it's in the 'sidebar_**group**_channel'.

jgilliam17 commented:
@devinbinnie I am getting the same errors on my macbook with Mojave
Maybe the trouble is with un-filtering unread channels on all the other teams during refresh
Can you try this:
1. Belong to at least 2 teams
2. While viewing one team - Engage the unread filter 
3. Click ⟳ to reload the page (reloads, filter disengages)
4. Click on the other team - page goes blank on Chrome, this is where I am consistently reproducing the error

jgilliam17 commented:
Question for the UX:
Now that channel switcher has been moved from the bottom of the sidebar, would it make sense to raise the `More Unreads ↓` toast up a bit, it is slightly obstructed by the bottom left corner channel url label

<img width="571" alt="Screen Shot 2020-03-13 at 5 52 24 PM" src="https://user-images.githubusercontent.com/52937121/76662098-705c2780-6553-11ea-97b2-5df1fc959a24.png">

<img width="320" alt="Screen Shot 2020-03-13 at 5 46 00 PM" src="https://user-images.githubusercontent.com/52937121/76661900-f0ce5880-6552-11ea-860b-451f112f1d67.png">

andrewbrown00 commented:
> Question for the UX:
> Now that channel switcher has been moved from the bottom of the sidebar, would it make sense to raise the `More Unreads ↓` toast up a bit, it is slightly obstructed by the bottom left corner channel url label

Absolutely @jgilliam17 good catch 👍

@devinbinnie can we move this up  so it's 8px above the url container

![image](https://user-images.githubusercontent.com/48559658/76761840-82fa7a80-6766-11ea-81ee-7ea719a0717c.png)

mattermod commented:
Test server destroyed

mattermod commented:
Creating a new SpinWick test server using Mattermost Cloud.

mattermod commented:
Mattermost test server created! :tada:

Access here: https://mattermost-webapp-pr-4999.test.mattermost.cloud

| Account Type | Username | Password |
|---|---|---|
| Admin | sysadmin | Sys@dmin123 |
| User | user-1 | User-1@123 |

jgilliam17 commented:
- Team creation hangs for me. It may be related to the issue I already mentioned above.
Team is created, but I had to exit this page to view.
<img width="1385" alt="Screen Shot 2020-03-16 at 10 16 01 AM" src="https://user-images.githubusercontent.com/52937121/76768971-2c933900-6772-11ea-985d-559393e50ef9.png">

- Desktop app crashed earlier [see crash report](https://docs.google.com/document/d/1BAorPmDdtjjaJlJS_5O0xiHUi9XJcs3YNS2_1BgsN6g/edit) when switching teams and trying to shift+cmd+r while on this screen below
<img width="1015" alt="Screen Shot 2020-03-16 at 9 03 33 AM" src="https://user-images.githubusercontent.com/52937121/76770926-fc996500-6774-11ea-848a-107aea796ed2.png">

mattermod commented:
New commit detected. SpinWick will upgrade if the updated docker image is available.

devinbinnie commented:
@jgilliam17 I've pushed up a fix for the team switching issue. Can you see if you can repro any crashes after the build is done?

devinbinnie commented on selectors/views/channel_sidebar.test.ts:
```diff
@@ -0,0 +1,48 @@
+        expect(Selectors.isCategoryCollapsed(store.getState(), category1)).toBe(false);
+        expect(Selectors.isCategoryCollapsed(store.getState(), category2)).toBe(false);
+    });
+});
```
Will add to ticket: https://mattermost.atlassian.net/browse/MM-20903

devinbinnie commented on components/status_icon_new.tsx:
```diff
@@ -0,0 +1,35 @@
+    status: string;
+}
+
+export default class StatusIconNew extends React.PureComponent<Props> {
```
I'm not sure what the final decision is as to using these icons in other parts of the app over just the sidebar. @andrewbrown00 can you provide some insight?

devinbinnie commented on components/sidebar/add_channel_dropdown/index.ts:
```diff
@@ -0,0 +1,24 @@
+        actions: bindActionCreators({
+        }, dispatch),
+    };
+}
```
Nope, it was there as a template. I'll remove this.

mattermod commented:
Mattermost test server updated with git commit `a88c838c47339920ea6ee825b50a09d3dbe451ab`.

Access here: https://mattermost-webapp-pr-4999.test.mattermost.cloud

mattermod commented:
New commit detected. SpinWick will upgrade if the updated docker image is available.

devinbinnie commented on components/sidebar/channel_filter/channel_filter.tsx:
```diff
@@ -0,0 +1,91 @@
+
+        return (
+            <div className='SidebarFilters'>
+                <a
```
I'm making the filter and navigator a region, and there's only up to 5 buttons underneath. Do we need to seperate them as sections? I would think not personally.

devinbinnie commented on components/sidebar/add_channel_dropdown/add_channel_dropdown.tsx:
```diff
@@ -0,0 +1,53 @@
+                    ariaLabel={formatMessage({id: 'sidebar_left.add_channel_dropdown.dropdownAriaLabel', defaultMessage: 'Add Channel Dropdown'})}
+                >
+                    <Menu.Group>
+                        <Menu.ItemAction
```
This is the same component that we use for the main menu, so it should support whatever that supports.

devinbinnie commented on components/sidebar/sidebar_channel/sidebar_channel_close/sidebar_channel_close.tsx:
```diff
@@ -0,0 +1,88 @@
+                    className='btn-close'
+                >
+                    {'×'}
+                </span>
```
Should I make a ticket?

deanwhillier replied:
Yes, please. This should be a button and/or an anchor tag. The structure will likely need to be updated though.

devinbinnie commented on components/sidebar/sidebar_channel/sidebar_channel_link/sidebar_channel_link.tsx:
```diff
@@ -0,0 +1,217 @@
+    //     } else {
+    //         this.setState({showTooltip: false});
+    //     }
+    // }
```
No, I've decided to just re-enable the code since this was missing.
I've left a comment because I feel like we need a better way to do this, perhaps CSS can help us hide stuff.

devinbinnie commented:
> Thanks for knocking out the other changes @devinbinnie 👍
> 
> A few more minor items to resolve:
> 
> * [ ]  The filter control tooltip shifts positions when toggling the filter

I think this is a bug with the `Tooltip` component, we should make a ticket to fix it seperately.

> 
> ![2020-03-16 10 05 29](https://user-images.githubusercontent.com/48559658/76766110-dde3a000-676d-11ea-92a7-276aab237181.gif)
> 
> * [x]   Switching teams gives the user a white screen (of death as I like to call it)

Should be fixed now.

> 
> ![2020-03-16 10 08 37](https://user-images.githubusercontent.com/48559658/76766336-361aa200-676e-11ea-8fc3-201b7e7f1e2e.gif)
> 
> * [x]  I can repro the theme flashing on top of the menu, it happens more after opening the menu several times

As mentioned before, I don't think there's anything I can do about this. The app thinks the `z-index` is different for a split-second. @deanwhillier any CSS insight I can employ here?

> 
> > While feature is ON, and user opens account settings menu or status menu, background theme can be seen/flashes on the menu. see gif
> 
> ![2020-03-16 10 17 23](https://user-images.githubusercontent.com/48559658/76767101-762e5480-676f-11ea-96d3-b2ac9225fbac.gif)
> 
> * [ ]  The keyboard shorts to navigate between channels is still acting funky and moves through hidden channels, instead of navigating through visible channels only when the unread filter is on.
> 
> ![2020-03-16 10 23 56](https://user-images.githubusercontent.com/48559658/76769239-901d6680-6772-11ea-97f6-4097bed0ec92.gif)

@hmhealey Is the selector not filtering out hidden channels?

mattermod commented:
Mattermost test server updated with git commit `8ede761dbdb38344851cdb0b034096add8132dd4`.

Access here: https://mattermost-webapp-pr-4999.test.mattermost.cloud

mattermod commented:
New commit detected. SpinWick will upgrade if the updated docker image is available.

mattermod commented:
Mattermost test server updated with git commit `25ac3a40c9b76c1b98d09cb70bd3d674c8086571`.

Access here: https://mattermost-webapp-pr-4999.test.mattermost.cloud

jgilliam17 commented:
Thanks @devinbinnie 
- Team switching white screen is still there (gray on desktop), same errors
- I am also unable to close DMs that I am not currently viewing 
e.g. try closing a DM with sysadmin, while viewing any other DM or channel

mattermod commented:
Test server destroyed

devinbinnie commented:
/update-branch

mattermod commented:
Test server creation failed. See the logs for more information.

mattermod commented:
New commit detected. SpinWick will upgrade if the updated docker image is available.

devinbinnie commented:
> Thanks @devinbinnie
> 
> * Team switching white screen is still there (gray on desktop), same errors

Can you do a full Empty Cache and Hard Reload once the server refreshes? If it's still happening, are you doing the same steps? I can't reproduce it locally anymore.

> * I am also unable to close DMs that I am not currently viewing
>   e.g. try closing a DM with sysadmin, while viewing any other DM or channel

@hmhealey any thoughts? The action for closing a channel seems to be working fine, but I'm not sure if the channel is rejoining or if it's not disappearing from the channel list.

mattermod commented:
Test server creation failed. See the logs for more information.

mattermod commented:
Creating a new SpinWick test server using Mattermost Cloud.

mattermod commented:
Mattermost test server created! :tada:

Access here: https://mattermost-webapp-pr-4999.test.mattermost.cloud

| Account Type | Username | Password |
|---|---|---|
| Admin | sysadmin | Sys@dmin123 |
| User | user-1 | User-1@123 |

mattermod commented:
New commit detected. SpinWick will upgrade if the updated docker image is available.

mattermod commented:
Mattermost test server updated with git commit `9a6b94ff5736c522b37d503cb01ffd76deb5ce90`.

Access here: https://mattermost-webapp-pr-4999.test.mattermost.cloud

mattermod commented:
New commit detected. SpinWick will upgrade if the updated docker image is available.

mattermod commented:
New commit detected. SpinWick will upgrade if the updated docker image is available.

mattermod commented:
Test server creation failed. See the logs for more information.

mattermod commented:
Mattermost test server updated with git commit `2129df2e23861ee77d33188c112490b0db75c7df`.

Access here: https://mattermost-webapp-pr-4999.test.mattermost.cloud

jgilliam17 commented:
- Team issue seems to be resolved, no more blank screen
- Tapping on + and then Browse Channels auto joins all available channels (populates LHS), doesn't remove them from the More Channels modal. Refreshing the page removes those channels that were auto-added.
<img width="1221" alt="Screen Shot 2020-03-16 at 2 45 06 PM" src="https://user-images.githubusercontent.com/52937121/76790613-9113bf80-6795-11ea-8b8d-eeb54f779f74.png">

jgilliam17 commented:
@devinbinnie re: DM closing issue
- DMs are now closing on a second click
DM is removed from the LHS after closing, but it doesn't redirect to Town Square, closed DM remains open on the main channel view

andrewbrown00 commented:
@devinbinnie you're right, the tooltip has a bug on controls that toggle, logged a bug https://mattermost.atlassian.net/browse/MM-23366

devinbinnie commented:
> @devinbinnie re: DM closing issue
> 
> * DMs are now closing on a second click
>   DM is removed from the LHS after closing, but it doesn't redirect to Town Square, closed DM remains open on the main channel view

I think @hmhealey has a fix for this in a different PR, so once that's merged that issue should be resolved.

jgilliam17 commented:
Devin few more issues
1. Desktop only - Missing tooltip on channels with long names
2. Back history arrow  - screen refreshes when user reaches the channel they first landed on.  Back arrow often stops at this channel, if you click few more times trying to advance further back in history you will lose the ability to use the forward arrow and sometimes get in a weird state where you can go forward but each forward will load white screen first then the channel. 

![forward refresh](https://user-images.githubusercontent.com/52937121/76864364-5498a000-6837-11ea-875a-ea99488a8156.gif)

4. Archived channels still remain on the LHS, I've seen some discussion about this earlier, are we waiting for a separate fix for this?

jgilliam17 commented:
2. May be just broken on release and daily when using cmd+[ and cmd+], I need to check again

devinbinnie commented:
> Devin few more issues
> 
> 1. Desktop only - Missing tooltip on channels with long names

This is a weird one, can we create a follow-up ticket for this? I've done some investigation and can't determine why this is happening.

> 2. Back history arrow  - screen refreshes when user reaches the channel they first landed on.  Back arrow often stops at this channel, if you click few more times trying to advance further back in history you will lose the ability to use the forward arrow and sometimes get in a weird state where you can go forward but each forward will load white screen first then the channel.
> 

There's unfortunately nothing we can do about this. The history function was originally intended to actually traverse a set channel history, but that proved to be very hard to do and was pushed from Phase 1. We elected to go with just using browser history, but unfortunately browsers don't provide JavaScript with the information of where you are in the history stack, so we can't determine if we should enable/disable those buttons. This should be fixed in a future phase when we move away from using the browser history.

> ![forward refresh](https://user-images.githubusercontent.com/52937121/76864364-5498a000-6837-11ea-875a-ea99488a8156.gif)
> 
> 1. Archived channels still remain on the LHS, I've seen some discussion about this earlier, are we waiting for a separate fix for this?

jgilliam17 commented:
for 1. - Added an issue for followup https://mattermost.atlassian.net/browse/MM-23384

mattermod commented:
New commit detected. SpinWick will upgrade if the updated docker image is available.

mattermod commented:
Mattermost test server updated with git commit `3bb10f5fc0bfb28c48458d90e87740a81b47e9dc`.

Access here: https://mattermost-webapp-pr-4999.test.mattermost.cloud

hmhealey commented:
> Archived channels still remain on the LHS, I've seen some discussion about this earlier, are we waiting for a separate fix for this?

That fix is in the redux version that should now be available here

hmhealey commented on components/sidebar/sidebar_category_list/index.ts:
```diff
@@ -0,0 +1,59 @@
+const getCategoriesForTeam = makeGetCategoriesForTeam();
+const getChannelsForCategory = makeGetChannelsForCategory();
+
+// TODO: Remove once Harrison's stuff is in
```
This can be replaced now

hmhealey commented on components/sidebar/sidebar_category_list/index.ts:
```diff
@@ -0,0 +1,59 @@
+import SidebarCategoryList from './sidebar_category_list';
+
+const getCategoriesForTeam = makeGetCategoriesForTeam();
+const getChannelsForCategory = makeGetChannelsForCategory();
```
These should also technically be instantiated per-component by moving them into a `makeMapStateToProps`, but I wouldn't worry about it for this one since there's only one instance of the sidebar

devinbinnie replied:
I'll move them anyway.

hmhealey commented on components/sidebar/sidebar_channel/sidebar_direct_channel/sidebar_direct_channel.tsx:
```diff
@@ -0,0 +1,143 @@
+            return null;
+        }
+
+        let displayName = channel.display_name;
```
Where are these extra props on the channel (namely `display_name` for DMs and `status`) coming from? The code that we're using now shouldn't be adding them to the channel object as far as I know

devinbinnie replied:
They are part of the `Channel` object type, I haven't added them in anywhere.

hmhealey replied:
The `display_name` should exist on the channel, but some of the other fields (`isCurrent`, `teammate_id`, `status`) are injected by some of the selectors as well as the `display_name` which is overridden for DMs/GMs. I don't think any of my selectors add those though, so I don't know where they'd be coming from.

hmhealey replied:
Oh, they're being injected by [`makeGetChannel`](https://github.com/mattermost/mattermost-redux/blob/master/src/selectors/entities/channels.ts#L112). I'm going to file a ticket to move away from that selector, but we can do that later.

hmhealey commented on components/sidebar/sidebar_category_list/sidebar_category_list.tsx:
```diff
@@ -0,0 +1,390 @@
+
+    renderCategory = (category: any) => {
+        return (
+            <SidebarCategory
```
Does this need a `key` prop? I think React will complain without one

hmhealey commented on components/sidebar/sidebar_category_list/sidebar_category_list.tsx:
```diff
@@ -0,0 +1,390 @@
+        }
+    };
+
+    renderCategory = (category: any) => {
```
Type?

hmhealey commented on components/sidebar/sidebar_channel/sidebar_channel.tsx:
```diff
@@ -0,0 +1,134 @@
+            >
+                <ChannelComponent
+                    channel={channel}
+                    currentTeamName={currentTeamName}
```
This could be left for a future improvement, but do we really need to pass this around everywhere? Those components are already connected to the store, so they can get it themselves. The previous code likely passed this around because it was rewritten in the period where we wanted less components connecting to the flux stores.

devinbinnie replied:
It's more about reducing the amount of times the store has to actually fetch it. I'm all for having redux grab it instead, but I'm not sure what's more performant. Let me know.

hmhealey replied:
Yeah, I'm not sure what's better. We've kind of gone back in forth on it since we don't know the overhead of `connect`. Best to leave it as-is for now

sudheerDev replied:
It's fine, Mostly a micro optimisation and not something we need to worry about IMO.  As long as we are not passing them between components which are not consuming it is fine.

hmhealey commented on components/sidebar/sidebar_channel/sidebar_channel_close/index.ts:
```diff
@@ -0,0 +1,33 @@
+    show: boolean;
+}
+
+function mapStateToProps(state: GlobalState, ownProps: OwnProps) {
```
Can you remove the reference to `ownProps`? Just having it defined actually causes `react-redux` to memoize differently

sudheerDev replied:
Do we even need to have this? Can't we pumm null to connect?

devinbinnie replied:
Yeah i'm going to remove the connector here as it isn't needed right now.

jgilliam17 commented:
> > Archived channels still remain on the LHS, I've seen some discussion about this earlier, are we waiting for a separate fix for this?
> 
> That fix is in the redux version that should now be available here

Thank you @hmhealey 
Currently, the archived channel is not shown on the LHS when open in the main channel view (contrary to existing functionality). Is this new and expected?

hmhealey commented on components/sidebar/sidebar_channel/sidebar_group_channel/index.ts:
```diff
@@ -0,0 +1,63 @@
+
+    let membersCount = 0;
+    if (memberIds && memberIds[ownProps.channel.id]) {
+        const groupMemberIds: Set<string> = memberIds[ownProps.channel.id] as unknown as Set<string>;
```
Why is this typing necessary? Is there some issue with the types from mattermost-redux? I know some of that is an outstanding issue, so I might've mentioned it before

devinbinnie replied:
Yeah I think the type in `mattermost-redux` is wrong, so I overrided here. We should make a ticket to fix this!

hmhealey commented on components/team_sidebar/team_sidebar.tsx:
```diff
@@ -0,0 +1,46 @@
+            root!.classList.remove('multi-teams');
+            return null;
+        }
+        root!.classList.add('multi-teams');
```
The class changes should happen in `componentDidMount`/`componentDidUpdate` and only when the number of teams changes

hmhealey replied:
Oh, this is how it works currently.... Gross.

Well, we can fix this when we get to actually rewriting this.

devinbinnie replied:
Yeah, this sidebar isn't even in use right now anyways.

jgilliam17 commented:
@devinbinnie 
On desktop, I noticed if you open `create team` modal at any point, and then use history arrow to go back, user will get stuck on this screen (this is not present on webapp or desktop using cmd+[ )

sudheerDev commented on components/sidebar/sidebar_category/sidebar_category.tsx:
```diff
@@ -0,0 +1,189 @@
+        return (
+            <SidebarChannel
+                channelId={channel.id}
+                setChannelRef={this.props.setChannelRef}
```
really minor Nit: if we destructure part of `props` at top we might as well destructure all available props used in the func

sudheerDev commented on components/sidebar/sidebar_category/sidebar_category.tsx:
```diff
@@ -0,0 +1,189 @@
+            }
+        }
+
+        let displayName = category.display_name;
```
Don't we wanna localize display names if they are not custom?

devinbinnie replied:
See the line below, we are doing that.

sudheerDev replied:
My bad. I read it wrong 😀

sudheerDev commented on components/sidebar/sidebar_category_list/index.ts:
```diff
@@ -0,0 +1,59 @@
+import SidebarCategoryList from './sidebar_category_list';
+
+const getCategoriesForTeam = makeGetCategoriesForTeam();
+const getChannelsForCategory = makeGetChannelsForCategory();
```
Is this intentional? This will effectively have only one instance of these functions and if we are using them for multiple components then memoization does not work.

sudheerDev commented on components/sidebar/sidebar_category_list/sidebar_category_list.tsx:
```diff
@@ -0,0 +1,390 @@
+
+        // TODO: Copying over so it doesn't get lost, but we don't have a design for the sidebar on mobile yet
+        // close the LHS on mobile when you change channels
+        if (this.props.currentChannel.id !== prevProps.currentChannel.id) {
```
Does this also need to be `this.props.currentChannel?.id`?

devinbinnie replied:
`currentChannel` isn't nullable, so really neither should have the `?` operator. I'll fix the other one.

sudheerDev commented on components/sidebar/sidebar_channel/channel_mention_badge/index.ts:
```diff
@@ -0,0 +1,28 @@
+    };
+}
+
+export default connect(mapStateToProps, mapDispatchToProps)(ChannelMentionBadge);
```
If we don't have any props we can pass in `null` here instead of empty object

devinbinnie replied:
I'm just going to unhook it from Redux altogether

sudheerDev commented on components/sidebar/sidebar_channel/index.ts:
```diff
@@ -0,0 +1,63 @@
+    };
+}
+
+function mapDispatchToProps(dispatch: Dispatch<GenericAction>) {
```
Why is this needed here?

devinbinnie replied:
It's not, was used as a template. I'll remove it.

sudheerDev commented:
@devinbinnie i was able to do only part of the review so far. Will go through the rest of the code tomorrow

sudheerDev commented on components/sidebar/sidebar_channel/sidebar_base_channel/sidebar_base_channel.tsx:
```diff
@@ -0,0 +1,93 @@
+                link={`/${currentTeamName}/channels/${channel.name}`}
+                label={channel.display_name}
+                ariaLabelPrefix={ariaLabelPrefix}
+                closeHandler={this.getCloseHandler()!}
```
Should this be just `closeHandler={this.getCloseHandler}`?

sudheerDev commented on components/team_sidebar/team_button/team_button.tsx:
```diff
@@ -0,0 +1,24 @@
+
+};
+
+export default class TeamButton extends React.PureComponent<Props, State> {
```
Nothing to act now but in general we can just use functions for components as small as this. React components are expensive in general and adding a new one for small ones like this can cause more harm than good.

devinbinnie replied:
Agreed, we're not currently using this component anyway but I plan to change a few of these components to FCs at some point.

sudheerDev commented:
@devinbinnie Completed my review on this.

hmhealey commented:
> Thank you @hmhealey
> Currently, the archived channel is not shown on the LHS when open in the main channel view (contrary to existing functionality). Is this new and expected?

Yeah, that's a known issue (https://mattermost.atlassian.net/browse/MM-23251) that we'll be addressing after this goes in

devinbinnie commented on actions/views/channel_sidebar.ts:
```diff
@@ -0,0 +1,30 @@
+}
+
+export function expandCategory(categoryId: string) {
+    // This would ideally remove the item, but that doesn't work for some reason
```
@hmhealey was originally using `removeItem` here, but the code literally wasn't working.
I'll clarify the comment a bit.

hmhealey replied:
Yeah, basically `removeItem` is broken. I think I filed a ticket for it.

devinbinnie commented on actions/views/channel_sidebar.ts:
```diff
@@ -0,0 +1,30 @@
+    return setItem(StoragePrefixes.CHANNEL_CATEGORY_COLLAPSED + categoryId, false);
+}
+
+export function setCategoryCollapsed(categoryId: string, collapsed: boolean) {
```
It's not actually a toggle, it allows you to explicitly set the collapsed state of the category. It's used as a toggle, but the logic is written to be a setter.

devinbinnie commented on components/sidebar/channel_navigator/index.ts:
```diff
@@ -0,0 +1,53 @@
+
+// TODO: For Phase 1. Will revisit history in Phase 2
+function goBack() {
+    return (dispatch: any, getState: any) => {
```
My assumption was they would be changed to use Redux, but since we decided to push that history bit to Phase 2, it didn't happen.

I'd like to leave these as is until we finalize the solution, but I will change them if need be.

sudheerDev replied:
Sure okay, Sounds fine.

hmhealey replied:
I'm a fan of doing it this way since it means that the code calling this doesn't need to be aware that the functionality for this will change in the future.

devinbinnie commented on components/sidebar/channel_navigator/index.ts:
```diff
@@ -0,0 +1,53 @@
+function mapStateToProps(state: GlobalState) {
+    return {
+        canGoBack: true, // TODO: Phase 1 only
+        canGoForward: true,
```
Correct.

jgilliam17 commented:
@devinbinnie few concerns 
1. alt+ arrow surfaces hidden channels one by one (when in collapsed state), I was expecting to skip over to next visible channel on the LHS
2. Desktop history arrows - you explained the limitations here, my concern is the white screen that occasionally appears as channels are slow to load 
3. It takes 2 clicks to Closed a DM. After closing, it is removed from the LHS, but remains open on main channel

mattermod commented:
New commit detected. SpinWick will upgrade if the updated docker image is available.

devinbinnie commented:
@sudheerDev @andrewbrown00 @hmhealey Please review ASAP.

@jgilliam17 I'll look into those issues and let you know when I have fixes for them :)

mattermod commented:
New commit detected. SpinWick will upgrade if the updated docker image is available.

mattermod commented:
Test server creation failed. See the logs for more information.

mattermod commented:
Mattermost test server updated with git commit `cde426f15897460e9262124c41e0cb7e15e69d0f`.

Access here: https://mattermost-webapp-pr-4999.test.mattermost.cloud

devinbinnie commented:
> @devinbinnie
> On desktop, I noticed if you open `create team` modal at any point, and then use history arrow to go back, user will get stuck on this screen (this is not present on webapp or desktop using cmd+[ )

@jgilliam17 That's a known issue with the Create Team modal, which is out of the scope of this PR. Feel free to file a bug ticket that we can address.

jgilliam17 commented:
> > @devinbinnie
> > On desktop, I noticed if you open `create team` modal at any point, and then use history arrow to go back, user will get stuck on this screen (this is not present on webapp or desktop using cmd+[ )
> 
> @jgilliam17 That's a known issue with the Create Team modal, which is out of the scope of this PR. Feel free to file a bug ticket that we can address.

Thanks Devin
I guess there is also a workaround and cmd+[ or ] can be used to move through history when user is stuck here.

mattermod commented:
New commit detected. SpinWick will upgrade if the updated docker image is available.

mattermod commented:
New commit detected. SpinWick will upgrade if the updated docker image is available.

devinbinnie commented:
@jgilliam17 I have fixed points 1 and 3, as for 2 there's still nothing we can do about it at this time. It's very hard to get stuck in a back state and the easy workaround is to refresh or use the hotkeys. Let me know if you have any other questions.

Please review ASAP. @hmhealey

mattermod commented:
Test server creation failed. See the logs for more information.

mattermod commented:
Mattermost test server updated with git commit `866899deb1b8d3df52fa50f240ed1a85cb5be2cf`.

Access here: https://mattermost-webapp-pr-4999.test.mattermost.cloud

hmhealey commented on selectors/views/channel_sidebar.ts:
```diff
@@ -18,3 +24,37 @@ export function isCategoryCollapsed(state: GlobalState, categoryId: string) {
+    const getChannelsByCategory = makeGetChannelsByCategory();
+
+    return createSelector(
+        (state: GlobalState) => state,
```
As discussed, this selector won't work because we have to pass the whole state into the result function which invalidates the whole thing.

mattermod commented:
New commit detected. SpinWick will upgrade if the updated docker image is available.

mattermod commented:
Mattermost test server updated with git commit `cf360a6628330f6c9e885a0a7eb31c07d60eccea`.

Access here: https://mattermost-webapp-pr-4999.test.mattermost.cloud

jgilliam17 commented:
> @jgilliam17 I have fixed points 1 and 3, as for 2 there's still nothing we can do about it at this time. It's very hard to get stuck in a back state and the easy workaround is to refresh or use the hotkeys. Let me know if you have any other questions.
> 
> Please review ASAP. @hmhealey

Thank you Devin!
1. Looks good in both webapp and desktop 
3. DMs close on one Click and are removed from the main view in the webapp, but not on desktop. Closed DM remains open on Desktop

devinbinnie commented:
> > @jgilliam17 I have fixed points 1 and 3, as for 2 there's still nothing we can do about it at this time. It's very hard to get stuck in a back state and the easy workaround is to refresh or use the hotkeys. Let me know if you have any other questions.
> > Please review ASAP. @hmhealey
> 
> Thank you Devin!
> 
> 1. Looks good in both webapp and desktop
> 2. DMs close on one Click and are removed from the main view in the webapp, but not on desktop. Closed DM remains open on Desktop

Fixed!

mattermod commented:
New commit detected. SpinWick will upgrade if the updated docker image is available.

mattermod commented:
New commit detected. SpinWick will upgrade if the updated docker image is available.

mattermod commented:
New commit detected. SpinWick will upgrade if the updated docker image is available.

mattermod commented:
Test server creation failed. See the logs for more information.

mattermod commented:
Test server creation failed. See the logs for more information.

mattermod commented:
Mattermost test server updated with git commit `8faa6e6f5088b1ca04807aa11f5af42e9802dce6`.

Access here: https://mattermost-webapp-pr-4999.test.mattermost.cloud

mattermod commented:
Test server destroyed

mattermod commented:
@devinbinnie
Error trying doing the automated Cherry picking. Please do this manually

```
+++ Updating remotes...
Fetching upstream
Fetching origin
+++ Updating remotes done...
+++ Creating local branch automated-cherry-pick-of-#4999-upstream-release-5.22-1584561918
Branch 'automated-cherry-pick-of-#4999-upstream-release-5.22-1584561918' set up to track remote branch 'release-5.22' from 'upstream'.
+++ Downloading patch to /tmp/4999.patch (in case you need to do this again)

+++ About to attempt cherry pick of PR. To reattempt:
  $ git am -3 /tmp/4999.patch

Applying: Moved out old sidebar and added setting to activate new one (#4553)
Using index info to reconstruct a base tree...
M	components/admin_console/admin_definition.jsx
M	components/needs_team/needs_team.jsx
M	components/sidebar/__snapshots__/sidebar.test.jsx.snap
M	components/sidebar/__snapshots__/sidebar_tutorial_tip.test.jsx.snap
M	components/sidebar/sidebar.jsx
M	components/sidebar/sidebar.test.jsx
M	components/sidebar/sidebar_channel/__snapshots__/sidebar_channel.test.jsx.snap
M	components/sidebar/sidebar_channel/sidebar_channel.test.jsx
M	components/sidebar/sidebar_channel_button_or_link/sidebar_channel_button_or_link.test.jsx
M	components/sidebar/sidebar_tutorial_tip.test.jsx
M	components/team_sidebar/team_sidebar_controller.jsx
M	components/user_settings/sidebar/__snapshots__/user_settings_sidebar.test.jsx.snap
M	components/user_settings/sidebar/user_settings_sidebar.jsx
M	components/user_settings/sidebar/user_settings_sidebar.test.jsx
M	i18n/en.json
Falling back to patching base and 3-way merge...
Auto-merging i18n/en.json
Auto-merging components/user_settings/sidebar/user_settings_sidebar.test.jsx
Auto-merging components/user_settings/sidebar/user_settings_sidebar.jsx
Auto-merging components/user_settings/sidebar/__snapshots__/user_settings_sidebar.test.jsx.snap
Removing components/sidebar/sidebar_channel_button_or_link/__snapshots__/sidebar_channel_button_or_link_icon.test.jsx.snap
Auto-merging components/needs_team/needs_team.jsx
Auto-merging components/legacy_team_sidebar/legacy_team_sidebar_controller.jsx
Auto-merging components/legacy_team_sidebar/index.js
Auto-merging components/legacy_sidebar/sidebar_tutorial_tip.test.jsx
CONFLICT (content): Merge conflict in components/legacy_sidebar/sidebar_tutorial_tip.test.jsx
Auto-merging components/legacy_sidebar/sidebar_channel_button_or_link/sidebar_channel_button_or_link_icon.test.jsx
Auto-merging components/legacy_sidebar/sidebar_channel_button_or_link/sidebar_channel_button_or_link_close_button.test.jsx
Auto-merging components/legacy_sidebar/sidebar_channel_button_or_link/sidebar_channel_button_or_link.test.jsx
Auto-merging components/legacy_sidebar/sidebar_channel_button_or_link/__snapshots__/sidebar_channel_button_or_link_close_button.test.jsx.snap
Auto-merging components/legacy_sidebar/sidebar_channel_button_or_link/__snapshots__/sidebar_channel_button_or_link.test.jsx.snap
Auto-merging components/legacy_sidebar/sidebar_channel/sidebar_channel.test.jsx
Auto-merging components/legacy_sidebar/sidebar_channel/__snapshots__/sidebar_channel.test.jsx.snap
Auto-merging components/legacy_sidebar/sidebar.test.jsx
Auto-merging components/legacy_sidebar/sidebar.jsx
CONFLICT (content): Merge conflict in components/legacy_sidebar/sidebar.jsx
Auto-merging components/legacy_sidebar/__snapshots__/sidebar_tutorial_tip.test.jsx.snap
Auto-merging components/legacy_sidebar/__snapshots__/sidebar.test.jsx.snap
Auto-merging components/admin_console/admin_definition.jsx
Patch failed at 0001 Moved out old sidebar and added setting to activate new one (#4553)
Use 'git am --show-current-patch' to see the failed patch
When you have resolved this problem, run "git am --continue".
If you prefer to skip this patch, run "git am --skip" instead.
To restore the original branch and stop patching, run "git am --abort".

+++ Conflicts detected:

UU components/legacy_sidebar/sidebar.jsx
UU components/legacy_sidebar/sidebar_tutorial_tip.test.jsx

+++ Aborting in-progress git am.

+++ Returning you to the master branch and cleaning up.

```

amyblais commented:
@devinbinnie Does this PR need to be cherry-picked?

devinbinnie commented:
@amyblais Nope, this was added to the re-cut v5.22 branches. It is on master and v5.22

metanerd commented:
/update-branch

mattermod commented:
Error trying to update the PR.
Please do it manually.
